DataGenerator/DataGenerator.py

```diff
@@ -1,216 +1,125 @@
-        if 'PET' in self.keys:
-            PETScanPath = sorted(path.glob(patient_id + '*' + self.config['ImageSession']['PET'] + '*'))
-            PETScanPath = Path(PETScanPath[0], 'scans')
-            PET_match_folder = sorted(PETScanPath.glob('*-PET_AX_SC'))
-            if len(PET_match_folder) < 1:
-                raise ValueError(self.PatientList[id].label + ' should have one match!')
-            full_PET_path = Path(PET_match_folder[0], 'resources', 'DICOM', 'files')
-
-            PETreader = sitk.ImageSeriesReader()
-            dicom_names = sorted(PETreader.GetGDCMSeriesFileNames(str(full_PET_path)))
-            PETreader.SetFileNames(dicom_names)
-            PETObj = PETreader.Execute()
-            PET = sitk.GetArrayFromImage(CTObj)
-            if 'mask_name' in self.config['DATA'] and (mask_img is not None):
-                datadict["PET"] = np.expand_dims(MaskCrop(PET, cropbox), 0)
-            else:
-                datadict["PET"] = np.expand_dims(PET, 0)
-
-            if self.transform:
-                try:
-                    transformed_data = self.transform(datadict["PET"])
-                except:
-                    print(self.PatientList[id].label + 'has transform problem.')
-
-                if transformed_data is None:
-                    datadict["PET"] = None
-                else:
-                    datadict["PET"] = torch.from_numpy(transformed_data)
-
-            # print(datadict["Anatomy"].size, type(datadict["Anatomy"]))
-        if "Clinical" in self.keys:
-            category_feat, numerical_feat = LoadClinicalData(self.config, [self.PatientList[id]])
-            n_category_feat = self.c_norm.transform(category_feat).toarray()
-            n_numerical_feat = self.n_norm.transform(numerical_feat)
-            data = np.concatenate((n_numerical_feat, n_category_feat), axis=1)
-            data = np.squeeze(data)
-            # data = clinical_data.iloc[id].to_numpy()
-            # num_data = self.n_norm.transform([numerical_data.iloc[id]])
-            # cat_data = self.c_norm.transform([category_data.iloc[id]]).toarray()
-            # data = np.concatenate((num_data, cat_data), axis=1)
-            datadict["Clinical"] = data
-
-        if (self.inference):
-            return datadict
-        else:
-            return datadict, np.float32(label)
-
+            llabel = self.PatientList.loc[i,self.target]
+            label = torch.as_tensor(label, dtype=torch.int64)
+            return data, label
 
 ### DataLoader
 class DataModule(LightningDataModule):
-    def __init__(self, PatientList, config, keys, train_transform=None, val_transform=None, batch_size=64, **kwargs):
+    def __init__(self, PatientList, train_transform=None, val_transform=None, batch_size=8, train_size=0.7, val_size=0.2, test_size=0.1,
+                 target="pCR",selected_channel=['CT','RTDose','Records'], targetROI='PTV',ROIRange=[60,60,20],

```

Comment:
Since you are passing the kwargs, you don't need to define them here (target, selected channels, ROIRange, etc..)

Reply:
Changed in the new commits

Reply:
Changed in the latest commit